### DISCUSSION TITLE
creating a resilient RTSP stream (fail over support)

### DISCUSSION THREAD

**Discussion by gractrit** [bluenviron/mediamtx#1676] [Questions]
I'm currently recording multiple RTSP camera streams in a classroom setting (directly from multiple cameras to disk with just ffmpeg).   Occasionally one of the RTSP streams will fail temporarily.  I'm looking to make the system more resilient. If I were to restart the ffmpeg recording, I would end up with multiple video files, and variable gaps in between. Joining the video clips together would result in a video that is shorter in runtime than the other camera streams.

I'm interested in using mediamtx (or maybe gortsplib?) to create a proxied RTSP stream that will continue to stream in the event that the source camera stream is unavailable.  The proxy would need to re-establish the source camera stream when it became available again.  During the "outage" time the client would continue to receive an uninterrupted stream, with something like just a blank screen for the duration of the source outage.  If, for example, I was recording for 60 minutes, and the source camera was down in the middle for 5 minutes, the final recording would still be 60 minutes, but with 5 minutes of a blank screen during the source camera outage.

Is anyone aware of a way to accomplish this, or is there an existing solution that would provide an RTSP stream with a "failover" mechanism?

Thanks.

**Comment by aler9** [comment]
this is a duplicate of https://github.com/bluenviron/gortsplib/discussions/239
